src_c/base.c

```diff
@@ -533,6 +533,124 @@ pg_TwoIntsFromObj(PyObject *obj, int *val1, int *val2)
     return 1;
 }
 
+static int
+pg_IntFromObjEx(PyObject *obj, int *val, const char *msg)
+{
+    if (PyFloat_Check(obj)) {
+        *val = (int)PyFloat_AS_DOUBLE(obj);
+        return 1;
+    }
+
+    *val = PyLong_AsLong(obj);
+    if (PyErr_Occurred()) {
+        if (msg && PyErr_ExceptionMatches(PyExc_TypeError)) {
+            PyErr_SetString(PyExc_TypeError, msg);
+        }
+        return 0;
+    }
+    return 1;
+}
+
+/* internal API only, assumes obj is a Sequence*/
+static int
+_pg_IntFromSeqIndexEx(PyObject *obj, int index, int *val, const char *msg)
+
+{
+    int result = 0;
+
+    PyObject *item = PySequence_ITEM(obj, index);
+    if (!item) {
+        return 0;
+    }
+    result = pg_IntFromObjEx(item, val, msg);
+    Py_DECREF(item);
+
+    return result;
+}
+
+static int
+pg_IntFromSeqIndexEx(PyObject *obj, int index, int *val, const char *msg)
+
+{
+    int result = 0;
+
+    if (PySequence_Check(obj)) {
+        PyObject *item = PySequence_ITEM(obj, index);
+        if (!item) {
+            return 0;
+        }
+        result = pg_IntFromObjEx(item, val, msg);
+        Py_DECREF(item);
+    }
+
+    return result;
+}
+
+static int
+pg_TwoIntsFromObjEx(PyObject *obj, int *val1, int *val2, const char *msg)
+{
+    Py_ssize_t length;
+    /*Faster path for tuples and lists*/
+    if (PyTuple_Check(obj) || PyList_Check(obj)) {
+        length = PySequence_Fast_GET_SIZE(obj);
+        PyObject **f_arr = PySequence_Fast_ITEMS(obj);
```

Comment:
According to this draft PEP:

https://peps.python.org/pep-0620/

This function is up for deprecation soon so we may want to either find a different solution now, or bear it in mind for the future.

Is this of any use:

https://peps.python.org/pep-0590/

From skimming some in depth discussion the general vibe I get is that the Vectorcall API is going to be preferred for this sort of thing by the Python devs into the future, but it is only added in 3.8 (with underscores of uncertainty) and 3.9 (for reals). There was some discussion about scrapping FASTCALL for a while - but now it is in the Limited API so who knows.